sdk/client-core/core/src/main/java/io/clientcore/core/implementation/util/Slf4jLoggerShim.java

```diff
@@ -0,0 +1,242 @@
+        }
+    }
+
+    public boolean canLogAtLevel(ClientLogger.LogLevel logLevel) {
+        if (logLevel == null) {
+            return false;
+        }
+
+        switch (logLevel) {
+            case VERBOSE:
+                return isVerboseEnabled;
+            case INFORMATIONAL:
+                return isInfoEnabled;
+            case WARNING:
+                return isWarnEnabled;
+            case ERROR:
+                return isErrorEnabled;
+            default:
+                return false;
+        }
+    }
+
+    public void performLogging(ClientLogger.LogLevel logLevel, String message, Throwable throwable) {
+        if (!canLogAtLevel(logLevel)) {
+            return;
+        }
+
+        if (slf4jLogger == null) {
+            // We've already included the exception stacktrace in the message, so there's no need to pass it again to
+            // the default logger. We'll still pass it to the SLF4J logger in case the provider wants to do something
+            // else with it.
+            defaultLogger.log(logLevel, message, null);
+
+            return;
+        }
+
+        try {
+            switch (logLevel) {
+                case VERBOSE:
+                    LOGGER_VERBOSE_METHOD_HANDLE.invoke(slf4jLogger, message, throwable);
+
+                    break;
+                case INFORMATIONAL:
+                    LOGGER_INFO_METHOD_HANDLE.invoke(slf4jLogger, message, throwable);
+
+                    break;
+                case WARNING:
+                    LOGGER_WARN_METHOD_HANDLE.invoke(slf4jLogger, message, throwable);
+
+                    break;
+                case ERROR:
+                    LOGGER_ERROR_METHOD_HANDLE.invoke(slf4jLogger, message, throwable);
+
+                    break;
+                default:
+                    // Don't do anything, this state shouldn't be possible.
+                    break;
+            }
+        } catch (Throwable e) {
+            defaultLogger.log(WARNING, "Failed to log message, SLF4J logging will be disabled.", e);
+
+            slf4jLogger = null;
+        }
+    }
+
+    private static Object createLogger(String className) {
+        if (LOGGER_FACTORY_GET_LOGGER_METHOD_HANDLE == null || NOP_LOGGER_CLASS == null) {
+            return null;
+        }
+
+        try {
+            Object logger = LOGGER_FACTORY_GET_LOGGER_METHOD_HANDLE.invoke(className);
+
+            if (NOP_LOGGER_CLASS.isAssignableFrom(logger.getClass())) {
+                DEFAULT_LOGGER.log(VERBOSE, "Resolved NOPLogger, SLF4J logging will be disabled.", null);
+
+                return null;
+            }
+
+            return logger;
+        } catch (Throwable e) {
+            DEFAULT_LOGGER.log(WARNING, "Failed to create SLF4J logger, SLF4J logging will be disabled.", e);
+
+            return null;
+        }
+    }
+
+    private static boolean isSlf4jEnabledAtLevel(Object logger, ClientLogger.LogLevel logLevel)
```

Comment:
I think a better design for this method would be taking the `logger` and returning all the log levels that are enabled. Right now, in the current design, if reflection isn't accessible, we're going to spam warning logs that the log level cannot be checked. Additionally, since this is speculative and the application will continue to run, these should be INFO or VERBOSE logs to reduce spam on customer logging systems.

Reply:
This method is called in the constructor only and caches the outcome in an instance variable.
Is the concern here that someone would change the code and start calling it per log?

Reply:
refactored a bit and this method no longer exist - the flags for supported levels are set in the constructor which eliminates a concern of someone calling the method for each log